File: 程序员面试金典第6版/面试题 01.03. String to URL LCCI.cpp
class Solution {
public:
    string replaceSpaces(string S, int length) {
        // int pos = S.length()-1;
        int pos = S.size()-1;
        for(int i=length-1; i>=0; i--) {
            if(S[i] == ' ') {
                S[pos--] = '0';
                S[pos--] = '2';
                S[pos--] = '%';
            } else {
                S[pos--] =  S[i];
            }
        }
        return S.substr(pos+1);
    }
};

File: 程序员面试金典第6版/面试题 01.03. String to URL LCCI.java
class Solution {
    public String replaceSpaces(String S, int length) {
        char[] chs = S.toCharArray();
        int i=length-1, j = S.length()-1;
        while(i >= 0) {
            if(chs[i] == ' ') {
                chs[j--] = '0';
                chs[j--] = '2';
                chs[j--] = '%';
            } else {
                chs[j--] = chs[i];
            }
            i--;
        }
        return String.valueOf(chs, j+1, S.length()-j-1);
    }
}